packages/decision-tree/src/hooks/usePagination/usePagination.ts

```diff
@@ -2,47 +2,83 @@ import { useReducer, useState, type Reducer } from 'react';
 import { useSteps } from '@cmpsr/components';
 import { PaginationResponse, PaginationState, PaginationAction, PaginationActions, PaginationProps } from './types';
 import { HandleAnswersActions } from '@hooks';
+import { sectionIntroId } from '../../DecisionTree.normalizer';
 
 export const usePagination = ({
   steps,
   initialState,
   answersDispatch,
   submitAnswer,
 }: PaginationProps): PaginationResponse => {
+  const firstStep = steps.findIndex(({ id }) => id === initialState.sectionId) ?? 0;
   const { activeStep, setActiveStep } = useSteps({
-    index: 0,
+    index: firstStep,
     count: steps.length,
   });
   const [pageHistory, setPageHistory] = useState<Array<PaginationState>>([]);
+  const [nextQuestionOverride, setNextQuestionOverride] = useState(initialState);
+
+  const setNextQuestion = (nextQuestion, answers) => {
+    const iSection = steps.findIndex((step) => step.id == nextQuestion.sectionId);
+    if (answers) answersDispatch({ type: HandleAnswersActions.SetPreviousAnswers, payload: answers });
+    answersDispatch({ type: HandleAnswersActions.ResetAnswer });
+    answersDispatch({ type: HandleAnswersActions.GetPreviousAnswer, payload: nextQuestion.questionId });
+    setActiveStep(iSection);
+
+    return {
+      currentQuestion: nextQuestion.questionId,
+      currentSection: nextQuestion.sectionId,
+    };
+  };
 
   const paginationReducer = (state: PaginationState, { type, payload }: PaginationAction) => {
     switch (type) {
       case PaginationActions.PreviousQuestion: {
         const { currentQuestion, currentSection, step } = pageHistory.at(-1);
+
+        if (currentQuestion.includes(sectionIntroId))
+          setNextQuestionOverride({ questionId: state.currentQuestion, sectionId: state.currentSection });
+        if (!currentQuestion.includes(sectionIntroId)) setNextQuestionOverride(null);
```

Comment:
if/else please, you are calling include two times which is not he must efficient

Reply:
you really like elses! I find it so strange

Reply:
you really like elses! I find it so strange